multipart-form-writer/progressbar.ml

```diff
@@ -0,0 +1,39 @@
+let erase_progress_bar size =
+  String.make size '\b'
+  |> print_string
+
+let display_progress_bar total_size current_progress =
+  let width = 80 in
+  let equal_count = (width - 10) * current_progress / total_size in
+  let progress = float_of_int(current_progress) /. float_of_int(total_size) *. 100.0 in
+  let pre_progress = "[" in
+  let progress_bar = String.make equal_count '=' in
+  let progress_padding = String.make (width - 10 - equal_count) ' ' in
+  let post_progress = Printf.sprintf "] %0.2f%%" progress in
+  let post_progress_padding = 
+      if total_size = current_progress then
+          (String.make (10 - 1 - String.length post_progress) ' ') ^ "\n"
+      else
+          String.make (10 - 1 - String.length post_progress) ' '
```

Comment:
`10 - 1 - String.length post_progress` should probably be extracted into a binding like `post_progress_padding_length` to avoid the repetition.